### PR TITLE
feat(server): dynamic model list from SDK (#686)

### PR DESCRIPTION
**Pull request by blamechris**
## Summary

- Replaces hardcoded `MODELS` constant with a mutable registry that auto-updates from the SDK's `supportedModels()` API after session initialization
- New models are broadcast to all connected clients via `available_models` WebSocket messages
- Adds `getModels()`, `updateModels()`, `resetModels()` API to `models.js` for runtime model management
- Falls back to hardcoded defaults until SDK reports available models

## Test plan

- [x] 23 unit tests for models.js (conversion, lookup rebuilds, edge cases)
- [x] 4 unit tests for sdk-session.js (_fetchSupportedModels flow)
- [x] 3 integration tests for ws-server.js (broadcast to clients, multi-client, missing data guard)
- [x] All 1018 existing tests pass
- [ ] Manual: start server, connect app, verify model list updates dynamically

### CODE REVIEW

**Review by blamechris — COMMENTED** [review]
## Agent Review: feat(server): dynamic model list from SDK (#686)

**Verdict: APPROVE** (0 critical, 2 suggestions, 1 nit)

### What This PR Does

Replaces the hardcoded `MODELS` constant with a mutable registry that auto-updates from the Agent SDK's `supportedModels()` API after session initialization. New models are broadcast to all connected clients via `available_models` WebSocket messages.

### Architecture Assessment

The design is clean:
- `models.js` gains `getModels()`, `updateModels()`, `resetModels()` — properly encapsulated mutation with `rebuildLookups()` keeping derived state (ALLOWED_MODEL_IDS, lookup maps) consistent.
- `sdk-session.js` fetches after `system.init`, emits `models_updated`. Non-blocking, error-swallowing.
- `session-manager.js` forwards as transient event (correct — not replayed on reconnect since the WS connect handler sends current models via `getModels()`).
- `ws-server.js` broadcasts to ALL clients (correct for global model list), handled in both multi-session and legacy-CLI paths.
- App already handles `available_models` messages — no app changes needed.

### Findings

#### S1: Dead import in ws-server.js (nit)
`MODELS` is still imported on line 12 but no longer referenced anywhere in the file — all usages were replaced with `getModels()`.

#### S2: _fetchSupportedModels called on every message (suggestion)
`_fetchSupportedModels()` fires on every `system.init`, which occurs per `sendMessage()` call. For a typical session with many messages, this means repeated API calls and broadcasts. Consider either:
- A debounce/cooldown (e.g., skip if fetched within the last 60 seconds)
- A `_modelsFetched` flag to only call once per session lifetime

Not a blocker — the API call is fast and errors are swallowed — but could reduce unnecessary chatter.

#### S3: sdk-session tests don't call resetModels() (suggestion)
The `_fetchSupportedModels` tests in `sdk-session.test.js` call `updateModels()` (indirectly via `session._fetchSupportedModels()`) which mutates the module-level state in `models.js`. No `resetModels()` cleanup in `afterEach`. Since Node.js test runner can run suites concurrently within a file, this could theoretically contaminate other tests that rely on the default model list. The `models.test.js` tests correctly use `beforeEach(() => resetModels())` — the sdk-session tests should do the same (import `resetModels` and call it in afterEach).

### Test Coverage

Excellent coverage: 23 unit tests for models.js, 4 for sdk-session, 3 integration tests for ws-server broadcasting. All 1018 tests pass (2 pre-existing tunnel integration failures, unrelated to this PR).

### Issues

- **Closes:** #686 (Make model list dynamic)
- **No from-review issues affected** by this PR

File: packages/server/src/sdk-session.js [hunk 1]
```diff
@@ -600,6 +602,25 @@ export class SdkSession extends EventEmitter {
...
+    try {
+      const sdkModels = await this._query.supportedModels()
+      const converted = updateModels(sdkModels)
+      if (converted && converted.length > 0) {
+        console.log(`[sdk-session] Dynamic model list: ${converted.map(m => m.id).join(', ')}`)
+        this.emit('models_updated', { models: converted })
+      }
+    } catch (err) {
+      console.warn(`[sdk-session] Failed to fetch supported models: ${err.message}`)
+    }
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/server/src/sdk-session.js]: Fixed in 6e99d876. Updated docstring from 'silently ignored' to 'logged and ignored' to match the actual console.warn behavior.

File: packages/server/tests/sdk-session.test.js [hunk 2]
```diff
@@ -488,6 +488,69 @@ describe('SdkSession', () => {
...
+    it('does not emit when no query is active', async () => {
+      const events = []
+      session.on('models_updated', (data) => events.push(data))
+
+      session._query = null
+      await session._fetchSupportedModels()
+
+      assert.equal(events.length, 0)
+    })
+  })
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/server/tests/sdk-session.test.js]: Fixed in 6e99d876. Added afterEach(() => resetModels()) to the _fetchSupportedModels describe block, imported resetModels from models.js.
